### PR TITLE
1495 2 app repository cluster role

### PR DESCRIPTION
**Pull request by absoludity**
Follows #1497 . Updates the app repository controller to be granted a clusterrole and clusterrolebinding to read AppRepositories.

I've used an easily identifiable and descriptive name for the clusterrole (rather than the templated `kubeapps.apprepository.fullname`), while ensuring it is unique per Kubeapps installation (appending the namespace).

Ref #1495

### CODE REVIEW

File: chart/kubeapps/templates/apprepository-rbac.yaml [hunk 1]
```diff
@@ -32,6 +32,18 @@ rules:
       - jobs
     verbs:
       - create
+---
+# Kubeapps can read and watch its own AppRepository resources cluster-wide.
```
Comment (andresmgot) [inline chart/kubeapps/templates/apprepository-rbac.yaml]: shouldn't we enable this behind a feature flag for the moment? that way we avoid "breaking" changes in the chart while the feature is being developed

Comment (absoludity) [reply chart/kubeapps/templates/apprepository-rbac.yaml]: I can put it behind a flag, but didn't see much point given that this doesn't affect existing behaviour, just enables the features which will themselves be behind a feature flag to work properly. Let me update.

Comment (andresmgot) [reply chart/kubeapps/templates/apprepository-rbac.yaml]: > given that this doesn't affect existing behaviour

well it affects in the sense that we are now installing a cluster wide resource (apart than the CRD)

Comment (absoludity) [reply chart/kubeapps/templates/apprepository-rbac.yaml]: PTAL :)

**Pull request merged**